packages/docusaurus-preset-classic/src/index.ts

```diff
@@ -74,13 +76,16 @@ export default function preset(
       makePluginConfig('@docusaurus/plugin-google-analytics', googleAnalytics),
     );
   }
-  if (debug || (debug === undefined && !isProd)) {
+  if (isDebugEnabled) {
     plugins.push(require.resolve('@docusaurus/plugin-debug'));
   }
   if (gtag) {
     plugins.push(makePluginConfig('@docusaurus/plugin-google-gtag', gtag));
   }
   if (isProd && sitemap !== false) {
+    if (isDebugEnabled) {
+      (sitemap.ignorePatterns ||= []).push(`/${debugPluginRouteBasePath}/**`);
```

Comment:
Nit: I don't really like relying on the return value of an assignment. I know I've personally written something like that to bypass TS checks:

https://github.com/facebook/docusaurus/blob/2c61bf25688cc9943d393cfd294ff5e72c63e7bf/packages/docusaurus-theme-translations/update.mjs#L212-L213

but if we have a choice, I'd rather be a bit more verbose like:

```suggestion
      sitemap.ignorePatterns ??= [];
      sitemap.ignorePatterns.push(`/${debugPluginRouteBasePath}/**`);
```

Purely stylistic though